app/components/avatar/avatar.rb

```diff
@@ -11,22 +11,30 @@ class Avatar < ApplicationComponent
       'fallback-squirrel.jpg'
     ].freeze
 
-    def initialize(url: nil, key: 0)
-      @url = url
-      @key = key
+    def initialize(user: nil, size: :large)
+      @user = user
+      @size = size
     end
 
     private
 
-    attr_reader :key
+    attr_reader :user, :size
+
+    def key
+      @key = user&.id || 0
+    end
 
     def url
-      @url || fallback_url
+      user&.avatar_url || fallback_url
     end
 
     def fallback_url
       file = FALLBACK_IMAGES[key % FALLBACK_IMAGES.length]
       "#{FALLBACK_BASE_URL}/#{file}"
     end
+
+    def css_class
```

Comment:
Once #42 is merged, you can simply do this:

```ruby
class Avatar < ApplicationComponent
  def initialize(user: nil, **)
    super

    ...
  end
end
```

And in the template:

```erb
<div class="<% class_names.join(' ') %>">
  ...
</div>
```

Or alternatively, by defining a `call` method on the component:

```ruby
class Avatar < ApplicationComponent
  ...
  
  def call
    content_tag(:div, ..., class: class_names)
  end
end
```